### ISSUE TITLE
serial.tools.miniterm returns OSError: [Errno 25] on Raspberry Pi OS

### ISSUE THREAD

**Issue by NicoCaldo** [pyserial/pyserial#613]
I'm trying to make my Rasberry Pi 4 with Raspberry Pi OS communicate with USB serial to my PC. The PC sees the COM port and the raspberry has the serial port visible **_ttyAMA0_**

When I launch from the terminal `sudo python3 -m serial.tools.miniterm` I got the error 

```
pi@raspberrypi:~ $ sudo python3 -m serial.tools.miniterm

--- Available ports:
---  1: /dev/ttyAMA0         'ttyAMA0'
--- Enter port index or full name: 1
--- Miniterm on /dev/ttyAMA0  9600,8,N,1 ---
--- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---
Exception in thread rx:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3/dist-packages/serial/tools/miniterm.py", line 445, in reader
    data = self.serial.read(self.serial.in_waiting or 1)
  File "/usr/lib/python3/dist-packages/serial/serialposix.py", line 467, in in_waiting
    s = fcntl.ioctl(self.fd, TIOCINQ, TIOCM_zero_str)
OSError: [Errno 25] Inappropriate ioctl for device
```

I have already activated the serial port on the Raspberry config and boot files

**Comment by NicoCaldo** [comment]
It turned out that instead of showing the correct port (which was ttyGS0 for USB-C) the command `sudo python3 -m serial.tools.miniterm` was showing only **ttyAMA0** which is the serial of the GPIO pins.

If you're in a similar situation where you don't know which port to use and the command above shows a port that doesn't work I suggest to launch from terminal **_ls /dev/tty*_** and try the lastest ports listed

**Comment by zsquareplusc** [comment]
so `/dev/ttyGS*` should be added to the "list ports" code, i guess?

**Comment by zsquareplusc** [comment]
implemented in d5bfd55418a27fe384912663ddfb930c4d00c014

**Comment by NicoCaldo** [comment]
@zsquareplusc it's been 2 weeks but the pip packet hasn't been updated...
